### PR TITLE
Beacon Node gRPC Server and Client Services

### PR DESCRIPTION
**Pull request by rauljordan**
Hi all,

This implements a beacon node gRPC server as a service alongside a gRPC client for a sharding client and beacon node to communicate effectively. Setting this up is important for us to deprecate the Sharding Manager Contract and update our client code to the latest spec. #315.

To get this working, just run the beacon node with the flag `--beacon-rpc-port 4000` and a sharding client with `--beacon-rpc-provider http://localhost:4000`.

### CODE REVIEW

File: beacon-chain/utils/flags.go [hunk 1]
```diff
@@ -26,4 +26,10 @@ var (
 		Name:  "pubkey",
 		Usage: "Validator's public key. Beacon chain node will listen to VRC log to determine when registration has completed based on this public key address.",
 	}
+	// BeaconRPCPort defines a beacon node RPC port to open.
+	BeaconRPCPort = cli.StringFlag{
+		Name:  "beacon-rpc-port",
```
Comment (terenc3t) [inline beacon-chain/utils/flags.go]: Let's think the name here, do we really need beacon? If a beacon code only opens up one kind of rpc port, it should just be `rpc-port` right?

File: beacon-chain/rpc/service.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"net"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+var log = logrus.WithField("prefix", "rpc")
+
+// Service hi.
```
Comment (terenc3t) [inline beacon-chain/rpc/service.go]: Is there a reason we put `hi` there? and for all the comments too

File: beacon-chain/utils/flags.go (hunk 1 above)
Comment (rauljordan) [reply beacon-chain/utils/flags.go]: Good point, will change

File: beacon-chain/rpc/service.go (hunk 2 above)
Comment (rauljordan) [reply beacon-chain/rpc/service.go]: Whoops, forgot to change lol

File: beacon-chain/main.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ VERSION:
 	app.Usage = "this is a beacon chain implementation for Ethereum 2.0"
 	app.Action = startNode
 
-	app.Flags = []cli.Flag{utils.SimulatorFlag, utils.VrcContractFlag, utils.PubKeyFlag, utils.Web3ProviderFlag, cmd.DataDirFlag, cmd.VerbosityFlag, debug.PProfFlag, debug.PProfAddrFlag, debug.PProfPortFlag, debug.MemProfileRateFlag, debug.CPUProfileFlag, debug.TraceFlag}
+	app.Flags = []cli.Flag{utils.SimulatorFlag, utils.VrcContractFlag, utils.PubKeyFlag, utils.Web3ProviderFlag, utils.BeaconRPCPort, cmd.DataDirFlag, cmd.VerbosityFlag, debug.PProfFlag, debug.PProfAddrFlag, debug.PProfPortFlag, debug.MemProfileRateFlag, debug.CPUProfileFlag, debug.TraceFlag}
```
Comment (terenc3t) [inline beacon-chain/main.go]: We should change `BeaconRPCPort` to `RPCPort` too

**Pull request merged**